crypto4pl.c

```diff
@@ -702,6 +704,26 @@ pl_crypto_password_hash(term_t tpw, term_t tsalt, term_t titer, term_t tdigest)
   return PL_unify_list_ncodes(tdigest, DIGEST_LEN, (char *) digest);
 }
 
+static foreign_t
+pl_crypto_password_hash_bcrypt(term_t tpw, term_t tsetting, term_t tdigest)
+{ char *pw, *setting;
+  size_t pwlen, settinglen;
+  const int DIGEST_LEN = 7 + 22 + 31 + 1;
+  char digest[DIGEST_LEN];
+
+  if ( !PL_get_nchars(tpw, &pwlen, &pw,
+                      CVT_ATOM|CVT_STRING|CVT_LIST|CVT_EXCEPTION|REP_UTF8) ||
+       !PL_get_nchars(tsetting, &settinglen, &setting,
+                      CVT_ATOM|CVT_STRING|CVT_LIST|CVT_EXCEPTION|REP_UTF8) )
+    return FALSE;
+
+  char* ret = _crypt_blowfish_rn(pw, setting, (char *) digest, DIGEST_LEN);
+  if ( ret == NULL )
```

Comment:
Exception?   Maybe PL_resource_error("memory") if this is the only reason for failure?

Reply:
As far as I can determine, `_crypt_blowfish_rn` returns null in the case of malformed `settings` or insufficient output buffer size. An exception might make more sense than returning false, but I don't know if "memory" is exactly right. Maybe `PL_domain_error("bcrypt_settings", setting)`?